hudi-client/src/main/java/org/apache/hudi/client/HoodieWriteClient.java

```diff
@@ -477,6 +497,8 @@ public HoodieRestoreMetadata restoreToInstant(final String instantTime) throws H
    */
   @Override
   public void close() {
+    AutoCleanerService.shutdownAutoCleaner(autoCleanerService);
+    autoCleanerService = null;
```

Comment:
this kind of resetting is probably needed after each write operation as well? may be its fine to just reinitialize the service after waitForCompletion.. food for thought..